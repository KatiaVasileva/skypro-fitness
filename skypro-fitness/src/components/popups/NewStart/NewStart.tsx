type NewStartProps = {
    onClick: () => void;
  };
  
  export const NewStart = ({
    onClick,
  }: NewStartProps) => {
    return (
      <div className="w-full h-full overflow-x-hidden fixed z-10 bg-dark-gray/50 top-0 left-0">
        <div className="block w-screen min-h-screen mx-auto my-0">
          <div className="h-screen flex items-center">
            <div className="flex flex-col items-center gap-[30px] bg-white w-[360px] min-h-[350px] shadow-[0px_4px_67px_-12px_rgba(0,0,0,0.13)] mx-auto my-0 px-[40px] py-[40px] rounded-[30px] border-[0.7px] border-solid border-[#d4dbe5]">
              <img
                src="/img/logo.png"
                alt="logo_modal"
                className="block w-[220px] h-[35px]"
              />
              <div className="h-[100px] w-[278px] text-center">
                <p className="text-2xl text-black font-medium mb-7">
                  Чтобы начать заново, добавьте курс на главной странице
                </p>
              </div>
              <button className="btn-primary h-[52px] w-[235px]" onClick={onClick}>
                На главную
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  };
  