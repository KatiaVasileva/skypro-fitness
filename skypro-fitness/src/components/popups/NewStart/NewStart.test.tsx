/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import * as router from "react-router";
import { BrowserRouter } from "react-router-dom";
import UserProvider from "../../../context/UserContext";
import CoursesProvider from "../../../context/CoursesContext";
import WorkoutProvider from "../../../context/WorkoutContext";
import { NewStart } from "./NewStart";

const navigate = jest.fn();

describe("NewStart component", () => {
  beforeEach(() => {
    jest.spyOn(router, "useNavigate").mockImplementation(() => navigate);
  });

  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <UserProvider>
          <CoursesProvider>
            <WorkoutProvider>
              <NewStart onClick={function (): void {
                            throw new Error("Function not implemented.");
                        } }/>
            </WorkoutProvider>
          </CoursesProvider>
        </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});