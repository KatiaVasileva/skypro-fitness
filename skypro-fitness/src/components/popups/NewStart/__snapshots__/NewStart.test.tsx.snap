// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewStart component renders correctly 1`] = `
<div>
  <div
    class="w-full h-full overflow-x-hidden fixed z-10 bg-dark-gray/50 top-0 left-0"
  >
    <div
      class="block w-screen min-h-screen mx-auto my-0"
    >
      <div
        class="h-screen flex items-center"
      >
        <div
          class="flex flex-col items-center gap-[30px] bg-white w-[360px] min-h-[350px] shadow-[0px_4px_67px_-12px_rgba(0,0,0,0.13)] mx-auto my-0 px-[40px] py-[40px] rounded-[30px] border-[0.7px] border-solid border-[#d4dbe5]"
        >
          <img
            alt="logo_modal"
            class="block w-[220px] h-[35px]"
            src="/img/logo.png"
          />
          <div
            class="h-[100px] w-[278px] text-center"
          >
            <p
              class="text-2xl text-black font-medium mb-7"
            >
              Чтобы начать заново, добавьте курс на главной странице
            </p>
          </div>
          <button
            class="btn-primary h-[52px] w-[235px]"
          >
            На главную
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
